### PR TITLE
84 simplify checksum in sync

### LINKED ISSUE #84: sync_data: combining `ignore_checksum = False` and `verify_checksum = True` always generates checksum mismatch error.
When executing `sync_data`, `ignore_checksum = False` prohibits checksum of iRODS objects being retrieved, causing the checksum verification (when `verify_checksum = True`) to compare the checksum of the downloaded file to `None`.

Two possible solutions:
+ Retrieve checksum on iRODS-objects when `verify_checksum = True`, regardless of the value of `ignore_checksum` (also requires changing the `__eq__` function of `FileObject`).
+ Do not accept the combination of `verify_checksum = True` and `ignore_checksum = False`.

### PR DESCRIPTION
**Pull request by maartenschermer**
Reduced the number of checksum parameters.

Fixes #84

### CODE REVIEW

File: tutorials/Data_sync.ipynb [hunk 1]
```diff
@@ -81,11 +82,10 @@
...
     "- The `max_level` option controls the depth up to which the file tree will be synchronized. With `max_level` set to None (default), there is no limit (full recursive synchronization). A max level of 1 synchronizes only the source's root, max level 2 also includes the first set of subfolders/subcollections and their contents, etc.\n",
     "- The `copy_empty_folders` (default False) option controls whether folders/collections that contain no files or subfolders/subcollections will be synchronized.\n",
     "- The `dry_run` option lists all the source files and folders that need to be synchronized without actually performing the synchronization.\n",
-    "- The `verify_checksum` (default True) option will calculate and verify the checksum on the data after up- or downloading. A checksum mismatch will generate an error, but will not abort the synchronization process."
+    "- The `verify_checksum` (default True) option will calculate and verify the checksum on the data after up- or downloading. A checksum mismatch will generate an error, aborting the synchronization process."
```
Comment (qubixes) [inline tutorials/Data_sync.ipynb:88]: Maybe it would be nice to give some instruction on how to deal with this situation?

**Review by qubixes — APPROVED** [review]
Nice, looks much more elegant this way!

**Review by chStaiger — CHANGES_REQUESTED** [review]
Much nicer!!! And indeed checksums should be always obeyed ;)

Can you please fix the integration tests? Then it is ready for approval.

**Comment by maartenschermer** [comment]
I fixed the tutorial and the integration test, but there seems to be a linting issue in the metadata. Shall I fix that too, or have they already been fixed in develop?

**Comment by qubixes** [comment]
> I fixed the tutorial and the integration test, but there seems to be a linting issue in the metadata. Shall I fix that too, or have they already been fixed in develop?

You can ignore the 3.8 linting errors they are fixed in #89

**Pull request merged**